### PR TITLE
Add example module for sofajraft

### LINKED ISSUE #1: Integrate `sofajraft` and write a simple example in `test` module
Write a simple example by using sofajraft in test module.

### PR DESCRIPTION
**Pull request by kairbon**
this close #1

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -12,4 +12,5 @@ target/
 *.bak
 *.log*
 data/
-jraft-core/src/org.dst.dmeta.test/resources/log4j2.xml
\ No newline at end of file
+jraft-core/src/org.dst.dmeta.test/resources/log4j2.xml
+tmp/
```
Comment (jovany-wang) [inline .gitignore]: Add a new line?

File: src/test/java/util/TestUtil.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+      jarDir = userDir.getParent() + File.separator + SUFFIX_JAR_DIR;
+    } else {
+      jarDir = userDir.getPath() + File.separator + SUFFIX_JAR_DIR;
+    }
+    for (int i = 0; i < NODE_NUM; i++) {
+      final List<String> startCommand = ImmutableList.of(
+          "java",
+          "-classpath",
+          jarDir,
+          "org.dst.dmeta.example.counter.CounterServer",
```
Comment (jovany-wang) [inline src/test/java/util/TestUtil.java]: Where is the code of `CounterServer`?

File: src/test/java/util/TestUtil.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+      // TODO(qwang): Refine this wait
+      nodeProcess[node].waitFor(1, TimeUnit.SECONDS);
+    } catch (Exception e) {
+      System.out.println(e.getMessage());
+      nodeProcess[node].destroy();
+      throw new RuntimeException("Error executing command " + String.join(" ", command), e);
+    }
+  }
+
+  public static void startAllnode() {
```
Comment (jovany-wang) [inline src/test/java/util/TestUtil.java]:
```suggestion
  public static void startAllNode() {
```

File: src/test/java/util/TestUtil.java (hunk 2 above)
Comment (kairbon) [reply src/test/java/util/TestUtil.java]: I added it when I created it.

**Comment by kairbon** [comment]
> Could you plz add a child module `test` and then put these code into `test` module.

OK. I will do it.

**Pull request merged**